src/dependency-manager/src/environment-config/base.ts

```diff
@@ -48,6 +26,9 @@ export abstract class EnvironmentConfig {
   getServiceDetails(key: string): EnvironmentService | undefined {
     const services = this.getServices();
     const ref = Object.keys(services).find(svc_key => key.startsWith(svc_key));
-    return ref ? services[ref] : undefined;
+    if (!ref) {
+      throw new Error(`Service ${key} not found in environment config`);
```

Comment:
This method can return undefined so we shouldn't throw an error. This will def break loading of deps that aren't defined in the env json.